### PR TITLE
Fix most flaky Python tests in `5.0-dev` branch

### PR DESCRIPTION
**Pull request by abidlabs**
Tests were failing because of (1) mistakes in adapting the `api_prefix` in the `Client` class, (2) breaking changes in `huggingface_hub` which changed the format of data returned from the Inference API (not all bugs have been fixed here, we'll need to check after 5.0 launch), or (3) tests that needed to be updated in light of new 5.0 API

A few are still failing in CI, but they are all passing for me locally. I can't figure out why but it might be because the CI is using an older version of `huggingface_hub`?

### CODE REVIEW

File: client/python/gradio_client/client.py [hunk 1]
```diff
@@ -1376,6 +1375,7 @@ def _upload_file(self, f: dict, data_index: int) -> dict[str, str]:
 
     def _download_file(self, x: dict) -> str:
         url_path = self.root_url + "file=" + x["path"]
+        print("URL PATH", url_path)
```
Comment (freddyaboulton) [inline client/python/gradio_client/client.py:1378]: Delete this

File: client/python/gradio_client/client.py [hunk 2]
```diff
@@ -156,7 +156,7 @@ def __init__(
         )
         api_prefix: str = self.config.get("api_prefix", "")
         self.api_prefix = api_prefix.lstrip("/") + "/"
-        self.src_prefixed = urllib.parse.urljoin(self.src, api_prefix) + "/"
+        self.src_prefixed = urllib.parse.urljoin(self.src, api_prefix).rstrip("/") + "/"
```
Comment (freddyaboulton) [inline client/python/gradio_client/client.py:159]: Why do we `rstrip` just to add back `/`?

**Review by freddyaboulton — APPROVED** [review]
Changes look good to me @abidlabs

File: client/python/gradio_client/client.py (hunk 1 above)
Comment (abidlabs) [reply client/python/gradio_client/client.py:1378]: thanks

File: client/python/gradio_client/client.py (hunk 2 above)
Comment (abidlabs) [reply client/python/gradio_client/client.py:159]: guarantees that the url ends in 1 and only 1 forward slash. previously was seeing some issues where sometimes there would be two forward slashes

**Pull request merged**